nsxt/resource_nsxt_transport_node.go

```diff
@@ -0,0 +1,1959 @@
+										"ip_assignment":           getIPAssignmentSchema("standard_host_switch.0.transport_node_profile_sub_config.0.ip_assignment"),
+										"uplink":                  getUplinksSchema(),
+									},
+								},
+							},
+							"name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Name of the transport node profile config option",
+							},
+						},
+					},
+				},
+				"transport_zone_endpoint": getTransportZoneEndpointSchema(),
+				"uplink":                  getUplinksSchema(),
+				"vmk_install_migration": {
+					Type:        schema.TypeList,
+					Description: "The vmknic and logical switch mappings",
+					Optional:    true,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"destination_network": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "The network id to which the ESX vmk interface will be migrated",
+							},
+							"device_name": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "ESX vmk interface name",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getTransportZoneEndpointSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Optional:    true,
+		Description: "Transport zone endpoints",
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"transport_zone_id": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Unique ID identifying the transport zone for this endpoint",
+				},
+				"transport_zone_profile_id": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Identifiers of the transport zone profiles associated with this transport zone endpoint on this transport node",
+					Elem: &schema.Schema{
+						Type: schema.TypeString,
+					},
+				},
+			},
+		},
+	}
+}
+
+func getUplinksSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"uplink_name": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "Uplink name from UplinkHostSwitch profile",
+				},
+				"vds_lag_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Link Aggregation Group (LAG) name of Virtual Distributed Switch",
+				},
+				"vds_uplink_name": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Uplink name of VMware vSphere Distributed Switch (VDS)",
+				},
+			},
+		},
+	}
+}
+
+func getHostSwitchProfileIDsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:        schema.TypeList,
+		Description: "Identifiers of host switch profiles to be associated with this host switch",
+		Optional:    true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"key": {
```

Comment:
Those are de-facto `type` and `id`. Should we consider flattening those profiles out (as we do in other resources) for user convenience?